controllers/namespacelabel/namespacelabel_controller.go

```diff
@@ -97,112 +81,84 @@ var (
+func New(mgr manager.Manager) *NamespaceLabelReconciler {
+	watchNS, err := resources.GetWatchNamespace()
 	if err != nil {
-		cancel()
-		return nil, errors.Wrap(err, "could not get watch namespace from namespacelabel controller")
+		panic("could not get watch namespace from namespacelabel controller")
 	}
 	namespaceSegments := strings.Split(watchNS, "-")
 	namespacePrefix := strings.Join(namespaceSegments[0:2], "-") + "-"
 	operatorNs := namespacePrefix + "operator"
 
-	return &ReconcileNamespaceLabel{
-		mgr:               mgr,
-		client:            mgr.GetClient(),
-		scheme:            mgr.GetScheme(),
+	return &NamespaceLabelReconciler{
+		Client:            mgr.GetClient(),
+		Scheme:            mgr.GetScheme(),
 		operatorNamespace: operatorNs,
-		context:           ctx,
-		cancel:            cancel,
 		log:               log,
-	}, nil
-}
-
-// add adds a new Controller to mgr with r as the reconcile.Reconciler
-func add(mgr manager.Manager, r reconcile.Reconciler) error {
-	// Create a new controller
-	c, err := controller.New("namespace-label-controller", mgr, controller.Options{Reconciler: r})
-	if err != nil {
-		return err
 	}
-
-	// Watch for changes to primary resource
-	err = c.Watch(&source.Kind{Type: &corev1.Namespace{}}, &handler.EnqueueRequestForObject{})
-	if err != nil {
-		return err
-	}
-
-	// Watch the for changes in config maps
-	err = c.Watch(
-		&source.Kind{Type: &corev1.ConfigMap{}},
-		&handler.EnqueueRequestForObject{},
-	)
-	if err != nil {
-		return err
-	}
-
-	return nil
 }
 
-// blank assignment to verify that ReconcileNamespace implements reconcile.Reconciler
-var _ reconcile.Reconciler = &ReconcileNamespaceLabel{}
-
-// ReconcileNamespaceLabel reconciles a namespace label object
-type ReconcileNamespaceLabel struct {
+// NamespaceLabelReconciler reconciles a namespace label object
+type NamespaceLabelReconciler struct {
 	// This client, initialized using mgr.Client() above, is a split client
 	// that reads objects from the cache and writes to the apiserver
-	client            k8sclient.Client
-	scheme            *runtime.Scheme
-	restConfig        *rest.Config
-	mgr               manager.Manager
+	k8sclient.Client
+	Scheme *runtime.Scheme
+
 	operatorNamespace string
 	controller        controller.Controller
-	context           context.Context
-	cancel            context.CancelFunc
 	log               l.Logger
 }
 
+// +kubebuilder:rbac:groups="",resources=namespaces,verbs=list;get;watch;update
+
 // Reconcile : The Controller will requeue the Request to be processed again if the returned error is non-nil or
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
-func (r *ReconcileNamespaceLabel) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+func (r *NamespaceLabelReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error) {
+	_ = context.Background()
+	// log := r.Log.WithValues("namespacelabel", request.NamespacedName)
```

Comment:
Log not needed anymore ? :thinking:

Reply:
That should've been removed. It's the default logger created when scaffolding with operator-sdk, but we removed that in favour of the internal logger implementations in the `resources` package. I'll remove the line